### DISCUSSION TITLE
Matching API prices with the bills

### DISCUSSION THREAD

**Discussion by rob0101** [amberelectric/public-api#32] [Q&A]
Anyone else tried matching their API sourced usage against their SA bills ?

I just tried to match up solar exports against my bills with no luck. Totalling the $ earned in each 30m comes out less than the total on the last bill.

I also found I need to add about 3.7c to every c/kwh to get from the spot price to the API usage price.  The bill line items suggest that number should be nearer 1.6c/kwh.

The SA Time Of Use categorisation of each 30m block looks correct in the API, but the bills show very very different kwh usage totals for each TOU period (I reported that one a couple of weeks ago).

**Comment by madpilot** [comment]
Ping info@amber.com.au with your account details, and I can look it up for you.

However, my first thought, especially if you are under, is our billing provider rounds negative exports (by negative I mean where you owe us money) to 0 - the API doesn't.

Check your feed and see how many intervals have a negative spotPrice

**Comment by madpilot** [comment]
Re your second point: I'll investigate when you send your account details across

**Comment by rob0101** [comment]
I believe I've found the cause of the discrepancy between my API (and mobile app) pricing and the billing.

Before my smart meter install I was billed 1.77c/kwh for "Environmental Certificate Cost"
This has not been charged on bills since the smart meter. The bill where old and smart meter are both shown only charges the ECC on the dates before the smart meter.

If I add it into my calculations I can map spot price to API/app price with a degree of accuracy.

**Reply by madpilot** [reply]
Interesting. I'll check with the finance team which is correct.

When you say "a degree of accuracy" are you applying your loss factor to your calculation?

**Reply by rob0101** [reply]
I do include the loss factor and my accuracy is now around 0.3c

**Reply by rob0101** [reply]
Correction. My accuracy is ~0.03c   which I could write off as rounding errors.

**Reply by madpilot** [reply]
Oh, yeah - that is totally rounding. Due to the terribleness of floats, you will see floating-point errors. Internally we use Decimals, but they don't serialize well, and it isn't really the point of this API to be billing correct.

**Reply by rob0101** [reply]
How is the API calculation for solar c/kwh computed ?

Instinctively it is..     (spot price  / 1.1 / solar loss factor) - solar fixed cost

The division by 1.1 is to remove the GST from the spot price.

For a spot price of 11.1089c (one I'm currently trying to work from) the above comes to 6.4526c
The API gives 7.23c

I'm expecting to be wrong on this one, as that formula would give lower API prices but to match the bill I need to find higher numbers.

**Reply by madpilot** [reply]
Ah, yeah - this old chestnut. 

As a generator, you get paid a component of the market fees. In SA, that component is 0.86c (inc GST).

`(7.23-6.4526)*1.1 = 0.855 ~ 0.86`, so if you use the formula `(spotPrice + 0.86) / 1.1 / lossFactor` you should be closer.

(I'm trying to get you a reference as to what the fee is exactly)

**Reply by madpilot** [reply]
Right, so this one is a bit tricky to explain, so bear with me.

There is a secondary market called Frequency Control Ancillary Services (FCAS), that gets called upon if the current AEMO bid stack can't supply enough power to meet demand. In this case, generators that can come online quickly (like gas plants or hydro) are called in to fill the gap.

We as a retailer pay for this as a separate bill (as it's a different market), however, some of that is offset by our customers that have solar and are effectively reducing our FCAS exposure. At the moment, we can't calculate that in real-time on a customer by customer basis, so we effectively average it for each state each year and give that fixed per kWh price back to our solar customers.

We are currently in the middle of re-calculating those numbers, so it'll change on July 1 for SA, NSW and QLD and August 1 for VIC.

TL;DR is a cost of business that we get charged that is reduced by our solar customers, so we try to pass that on as best we can.
